share/availability/light/availability.go

```diff
@@ -103,7 +103,7 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 		}
 		// Verify total samples count.
 		totalSamples := len(samples.Remaining) + len(samples.Available)
-		if totalSamples != int(la.params.SampleAmount) {
+		if (totalSamples != int(la.params.SampleAmount)) && (totalSamples != len(dah.RowRoots)*len(dah.RowRoots)) {
```

Comment:
Imagine if sample la.params.SampleAmount is not 16. We should detect only when params value is higher when eds size. 
```suggestion
		if (totalSamples != int(la.params.SampleAmount)) && (la.params.SampleAmount > len(dah.RowRoots)*len(dah.RowRoots)) {
```

Reply:
I was thinking about this, but I also didn't think it was a problem if we sample *more* -- only if we sample less.